src/dft/backends/cufft/commit.cpp

```diff
@@ -323,6 +350,77 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         return offsets;
     }
 
+    virtual void set_workspace(scalar_type* usm_workspace) override {
+        this->external_workspace_helper_.set_workspace_throw(*this, usm_workspace);
+        if (plans[0]) {
+            cufftSetWorkArea(*plans[0], usm_workspace);
+        }
+        if (plans[1]) {
+            cufftSetWorkArea(
+                *plans[1], usm_workspace + get_plan0_workspace_size_bytes() / sizeof(scalar_type));
+        }
+    }
+
+    void set_buffer_workspace(cufftHandle plan, sycl::buffer<scalar_type>& buffer_workspace,
+                              std::size_t range, std::size_t offset) {
+        this->get_queue()
+            .submit([&](sycl::handler& cgh) {
+                auto workspace_acc =
+                    buffer_workspace.template get_access<sycl::access::mode::read_write>(cgh, range,
+                                                                                         offset);
+                cgh.host_task([=](sycl::interop_handle ih) {
+                    auto stream = ih.get_native_queue<sycl::backend::ext_oneapi_cuda>();
+                    auto result = cufftSetStream(plan, stream);
+                    if (result != CUFFT_SUCCESS) {
+                        throw oneapi::mkl::exception(
+                            "dft/backends/cufft", "set_workspace",
+                            "cufftSetStream returned " + std::to_string(result));
+                    }
```

Comment:
Why are you setting the stream? the stream is changed with each call to compute forward

Reply:
No longer calling setStream in [bee7d30](https://github.com/oneapi-src/oneMKL/pull/411/commits/bee7d3068a28fa90a868f23baab198dffad3440b).